circle.yml

```diff
@@ -1,11 +1,23 @@
 machine:
   services:
     - redis
-    - elasticsearch
   environment:
     REDIS_URL: redis://localhost:6379
     JASMINE_CONFIG_PATH: test/javascripts/support/jasmine.yml
 
-checkout:
+dependencies:
+  cache_directories:
+    - elasticsearch-2.4.6
+
+  post:
+    - if [[ ! -e elasticsearch-2.4.6 ]]; then wget https://download.elastic.co/elasticsearch/release/org/elasticsearch/distribution/tar/elasticsearch/2.4.6/elasticsearch-2.4.6.tar.gz && tar -xvf elasticsearch-2.4.6.tar.gz; fi
+    - elasticsearch-2.4.6/bin/elasticsearch: {background: true}
+    - sleep 10 && wget --waitretry=5 --retry-connrefused -v http://127.0.0.1:9200/
+
+database:
+  post:
+    - bin/rails search:rebuild
```

Comment:
The newer Elasticsearch gems caused issues where tests would fail while creating posts, threads, etc. because indexes didn't exist yet. This creates the indexes before running the tests.